docs/docs/specs/v0/ui-system.md

```diff
@@ -0,0 +1,41 @@
+# UI System
+
+Here is how the UI system is organized into separate concerns:
+
+## UI Primitives
+> Lookless, pure components.
+@dxos/react-{…}
+packages/ui/primitives/react-{…}
+
+## Aurora UI
+> Look-ful, pure components.
+@dxos/aurora
+packages/ui/aurora
+
+> theme elements and tokens that can be applied to primitives
+@dxos/aurora-theme
+packages/ui/aurora-theme
+
+## Aurora X
+> Other big pieces that are congruent with Aurora and separately packaged. e.g.: aurora-composer, aurora-datagrid.
+> These are always provided in pure form, and with an "optional" ECHO adapter if necessary.
+@dxos/aurora-* 
+packages/ui/aurora-*
+
+## Shell
+> HALO Button and shell panels and popovers
+@dxos/react-shell
+packages/sdk/react-shell
+
+## Shared 1P libs:
+> Error boundaries and telemetry harnesses, logos and other visual and non-visual things shared between 1P apps
+@braneframe/react-appkit
+packages/ui/react-appkit
+
+## Braneframe (Composer v2)
+@braneframe/app
+packages/apps/braneframe-app
+
+## Braneframe plugins
```

Comment:
in this model, Plugins are a new concpt which can contain and provide any number of elements that satisfy the various extensibility surfaces of the application. In my mind, Frames were ways of delegating the presentation of a rectangle to an extension developer. I think the right interface for that is `React.FC<T>` and a well known `Context` with hook such as `useSpecialContext()`. Depending on the surface where that's done, the plugin API for providing that rectangle can be similar but can also be unique (as is often necessary: i.e. the different mode Frames required when going full screen vs embedded in a document/stack). In other words, Frames are just React Components which can assume certain props and context.

Reply:
There's part of this I really like. Your plugin abstraction rhymes with the old MS IUnknown API (for OLE/COM/DCOM), except I'm not sure if you intend for this to be decoupled. I think the common theme here is my need for us to: a) move incrementally from what we currently have; b) from the outset define the modular extensibility (i.e., unit of deploy/assembly -- the linkage of the existing declarative module metadata to the reusable implementation.

In both of these cases, we can change what we currently have, as long as it moves both Composer and Kai in the same direction -- so if we change or replace some existing concepts, then we need to retrofit that to both existing apps preserving the current functionality -- otherwise they will continue on separate paths and impede our velocity.